jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/actions/CaseSimpleActionListViewImpl.java

```diff
@@ -0,0 +1,123 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.workbench.cm.client.actions;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import org.jboss.errai.common.client.dom.Div;
+import org.jboss.errai.common.client.dom.HTMLElement;
+import org.jboss.errai.common.client.dom.Span;
+import org.jboss.errai.databinding.client.api.DataBinder;
+import org.jboss.errai.databinding.client.components.ListComponent;
+import org.jboss.errai.ui.client.local.spi.TranslationService;
+import org.jboss.errai.ui.shared.api.annotations.AutoBound;
+import org.jboss.errai.ui.shared.api.annotations.Bound;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+import org.jbpm.workbench.cm.client.util.AbstractView;
+import org.jbpm.workbench.cm.model.CaseActionSummary;
+
+import static org.jboss.errai.common.client.dom.DOMUtil.*;
+import static org.jbpm.workbench.cm.client.actions.CaseActionListsViewImpl.*;
+
+
+@Dependent
+@Templated(stylesheet = "CaseSimpleActionListViewImpl.css")
+public class CaseSimpleActionListViewImpl extends AbstractView<CaseActionListsPresenter> implements CaseActionListsPresenter.CaseSimpleActionListView {
+
+    @Inject
+    @DataField("simple-list")
+    Div simpleList;
+
+    @Inject
+    @DataField("actions-list-header-image")
+    Span actionsListHeaderImage;
+
+    @Inject
+    @DataField("actions-list-header-text")
+    Span actionsListHeaderText;
+
+    @Inject
+    @DataField("empty-list-item")
+    private Div emptyContainer;
+
+    @Inject
+    @Bound
+    @DataField("actions-list")
+    private ListComponent<CaseActionSummary, CaseActionItemView> tasks;
+
+    @Inject
+    @AutoBound
+    private DataBinder<List<CaseActionSummary>> caseActionList;
+
+    @Inject
+    private TranslationService translationService;
+
+    @Override
+    public void init(final CaseActionListsPresenter presenter) {
+        super.init(presenter);
+        tasks.addComponentCreationHandler(v -> v.init(presenter));
+    }
+
+    public void setCaseActionList(final List<CaseActionSummary> caseActionList) {
+        this.caseActionList.setModel(caseActionList);
+        if (caseActionList.isEmpty()) {
+            removeCSSClass(emptyContainer, "hidden");
+        } else {
+            addCSSClass(emptyContainer, "hidden");
+        }
+    }
+
+    @Override
+    public void removeAllTasks() {
+        caseActionList.setModel(new ArrayList<>());
+    }
+
+    public void updateListHeader(String headerType) {
```

Comment:
I guess instead of creating String constants and passing as parameter, you could create a specific method for each action type to handle the changes needed.